nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -238,10 +262,46 @@ in {
       '';
     };
 
+    secureBoot = mkOption {
+      default = {};
+      type = types.submodule ({ config, ... }: {
```

Comment:
Why is this a submodule instead of just individual options like `secureBoot.enable`?

Reply:
No good reason indeed

Reply:
I guess the submodule idea is to have an interface we can move somewhere else and that defines what does it mean to enable Secure Boot.